contracts/CryptoMafiaCoin.sol

```diff
@@ -4,12 +4,45 @@ pragma solidity ^0.8.0;
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 
+interface RandomInterface {
+  function getRandomNumber(uint256 userProvidedSeed) external returns (bytes32);
+  function randomNumber() external view returns(uint256);
+}
+
+interface AccountInterface{
+    function ownerOf(uint256 tokenId) external view returns (address);
+}
+
 contract CryptoMafiaCoin is ERC20, Ownable {
     address private _accountsHolder;
-
-    constructor(address accountsHolder, uint256 initialSupply) ERC20("Crypto Mafia Coin", "CMC") {
+    
+    address addressRandomGenerator;
+    
+    address contractAddress;
+    
+    RandomInterface RNG;
+    
+    AccountInterface AI;
+    
+    mapping(uint256=>uint256) roomStake;
+    mapping(uint256=>mapping(uint256=>address)) playerID;
+    mapping(uint256=>uint256) roomRandomNumber;
+    mapping(uint256=>bool) randomNumIsTaken;
+    
+    constructor(address accountsHolder, uint256 initialSupply) ERC20("Crypto Mafia Coin Test", "CMC") {
         _mint(_msgSender(), initialSupply);
         _accountsHolder = accountsHolder;
+        contractAddress = address(this);
+        RequestRandomNumberCMC(0);
+    }
+    
+    
+    function setAccountAddress(address accountAddress) public {
+        AI = AccountInterface(accountAddress);
+    }
+    
+    function setRandomAddress(address accountAddress) public {
+        RNG = RandomInterface(accountAddress);
     }
 
```

Comment:
Imagine a random person changing the addess and causing contract to misbehave. In worst case hack the contract and drain the fund

Reply:
I'll change it to onlyOwner